Creatbot F430 config file

Drury13 opened the pull request:
this is a working config file for the Creatbot F430 using V9.5 mainboard.
can be used as a base for other Creatbot printers that use the same mainboard

Drury13 commented on config/Creatbot F430 V9.5.cfg:
```diff
@@ -0,0 +1,254 @@
+    SET_GCODE_OFFSET X=36
+    ACTIVATE_EXTRUDER EXTRUDER=extruder1
+
+
```
## Heaters ##

ld3300 commented on config/Creatbot F430 V9.5.cfg:
```diff
@@ -0,0 +1,254 @@
+microsteps: 16
+rotation_distance: 40.64
+endstop_pin: ^!PC0
+#endstop_pin: ^PE4
```
Can you add a comment why this line is commented (different board config) or remove it if unneeded

ld3300 commented on config/Creatbot F430 V9.5.cfg:
```diff
@@ -0,0 +1,254 @@
+z_hop: 10                 # Move up 10mm
+z_hop_speed: 5
+
+[bed_mesh]
```
bed_mesh should probably be removed, as it would be an optional addition for someone's config.

ld3300 commented:
Hi @Drury13 ,
I just had a couple of code questions listed above.  Thank you for doing the work to add this to the config.  I don't have the printer, so can't confirm anything, but if you have any questions please let me know.  You should just be able to make changes to the branch and push them to update this pull request.

Please edit your original commit comment to add the required "Signed-off-by:" line so there won't be any delays merging your commit.  See [here](https://www.klipper3d.org/CONTRIBUTING.html?h=signe#format-of-commit-messages)

Drury13 commented:
hi i have updated this, added notes on additional endstop pins, removed fan section.
i have left the bed_mesh section as i feel it would be more useful to leave it in plus it can be ignored if not wanted

KevinOConnor commented:
Thanks, however this config would need to comply with the guidelines at https://www.klipper3d.org/Example_Configs.html .  At a minimum, points 4, 7c, 7f.  Also, the advice to use 115200 baud is incorrect - all 3d printers Klipper currently supports work well at the default 250000.  Also, the rotation_distance for stepper_x and stepper_y look incorrect - see https://www.klipper3d.org/Rotation_Distance.html#using-a-gear_ratio

-Kevin

Drury13 commented:
Hi Kevin, i have renamed the config added to printers.test and removed save config section. i cant see where 7f applies.
the baud rate was taken from the original marlin config and has been running fine, i can test it at 250000 next week if you would like.
the rotation distance is tuned and correct, the printer uses MXL belts not GT2 which is why the strange number

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+#  Processor model: atmega2560
+#  Processor speed: 16Mhz
+#  Communication: UART0
+#  Baud: 152000
```
Should be default 250000

Drury13 replied:
see reply above

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+
+[mcu]
+serial: /dev/ttyACM0
+baud: 115200
```
Should be default 250000

Drury13 replied:
see reply above

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+max_accel: 500
+max_z_velocity: 5
+max_z_accel: 100
+square_corner_velocity: 5.0
```
Not needed. Default

Drury13 replied:
which bit?

Drury13 replied:
removed

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+dir_pin: !PH0
+enable_pin: !PL1
+microsteps: 16
+rotation_distance: 40.64
```
Rotation distance looks strange. Should adhere to hardware spec, i.e. belt pitch, teeth count etc

Drury13 replied:
see reply above

Sineos replied:
Ok, seems correct for MXL belt and 20 teeth pulley

Sineos replied:
Maybe a comment that it is a MXL belt and the respective teeth count would be good.

Drury13 replied:
added comment about MXL belts to top. not with printer to verify tooth count

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+dir_pin: !PF0
+enable_pin: !PF1
+microsteps: 16
+rotation_distance: 60.96
```
Rotation distance looks strange. Should adhere to hardware spec, i.e. belt pitch, teeth count etc

Drury13 replied:
see reply above

Sineos replied:
30 teeth pulley?

Drury13 replied:
added comment about MXL belts to top. not with printer to verify tooth count

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+enable_pin: !PA2
+microsteps: 16
+rotation_distance: 22.5
+full_steps_per_rotation: 200
```
Not needed. Default

Drury13 replied:
removed

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+enable_pin: !PA5
+microsteps: 16
+rotation_distance: 22.5
+full_steps_per_rotation: 200
```
Not needed. Default

Drury13 replied:
which bit?

Sineos replied:
Comment always apply to the line directly above

Drury13 replied:
removed

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+#pid_Ki: 0.76
+#pid_Kd: 55
+min_temp: 0
+max_temp: 400
```
Does 2nd extruder have a different max_temp compared to 1st?

Drury13 replied:
yes it does

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+pid_Kp: 54.027
+pid_Ki: 0.770
+pid_Kd: 948.182
+#pwm_cycle_time:
```
Remove

Drury13 replied:
all of it or just the commented bit?

Sineos replied:
Just the commented line.

Drury13 replied:
removed

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+max_temp: 75
+
+[verify_heater chamber_heater]
+max_error: 120
```
Not needed. Default

Drury13 replied:
removed

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+[verify_heater chamber_heater]
+max_error: 120
+check_gain_time: 120
+hysteresis: 5
```
Not needed. Default

Drury13 replied:
removed

Sineos commented on config/printer-creatbot-f430-v9.5.cfg:
```diff
@@ -0,0 +1,245 @@
+mesh_max: 380, 275
+probe_count: 5,5
+algorithm: bicubic
+relative_reference_index: 12
```
Is this needed? Typically not set when using a probe as Z endstop as it overrides the mesh z-offset

Drury13 replied:
Z endstop only probes the same point and if the bed is not flat and need a mesh?

Sineos replied:
See https://www.klipper3d.org/Bed_Mesh.html?#the-relative-reference-index

Typically only relevant if you have a physical endstop for the general homing and a probe for bed mesh. If you use the probe for both functionalities it is not needed. Creating and using a bed mesh works anyway.

Drury13 replied:
sorry do you just mean line 233 and not the whole bed mesh section?

Sineos replied:
Only relative_reference_index

Drury13 replied:
ok i will remove it. any idea why the white space check is failing?

Sineos replied:
Looks like windows line endings instead of unix

Sineos commented:
Many thanks for your support @Drury13 
Points I see open for now:
* Verify BAUD rate
* Modify line ending from Windows (CRLF) to Unix (LF) to pass the build test
* Add a signed-off line as per https://www.klipper3d.org/CONTRIBUTING.html#format-of-commit-messages

Sineos commented:
Git can be a pain and I'm by no means a Git expert. Maybe try:

```
git config --global core.eol lf
git config --global core.autocrlf false
```

Then set your cfg to LF in your editor, save and hope it will pick it up.
